lib/node_modules/@stdlib/number/float32/base/normalize/test/test.assign.js

```diff
@@ -0,0 +1,186 @@
+	var frac;
+	var exp;
+	var out;
+	var x1;
+	var x;
+	var v;
+	var i;
+
+	// Smallest denormalized number:
+	out = [ 0.0, 0.0 ];
+	v = normalizef( FLOAT32_SMALLEST_SUBNORMAL, out, 1, 0 );
+	t.equal( v, out, 'returns output array' );
+	t.ok( v[ 0 ] >= FLOAT32_SMALLEST_NORMAL, 'returns a normal number' );
+	t.equal( v[ 0 ]*pow( 2.0, v[ 1 ] ), FLOAT32_SMALLEST_SUBNORMAL, 'x = y * 2^exp' );
+
+	// Other subnormals...
+	for ( i = 0; i < 1000; i++ ) {
+		frac = 0.26 + (randu()*10.0); // 0.26 prevents underflow
+		exp = -38 - round( randu()*6.0 );
+		x = frac * pow( 10.0, exp );
+		x = toFloat32( x );
+
+		out = [ 0.0, 0.0 ];
+		v = normalizef( x, out, 1, 0 );
+		t.equal( v, out, 'returns output array' );
+		t.ok( v[ 0 ] >= FLOAT32_SMALLEST_NORMAL, 'returns a normal number ' + v[0] );
+
+		x1 = v[ 0 ] * pow( 2.0, v[ 1 ] );
+		x1 = toFloat32( x1 );
+		t.equal( x1, x, 'y*2^exp=x. y='+v[0]+', exp='+v[1]+', x='+x );
+	}
+	t.end();
+});
+
+tape( 'the function returns `[0,0]` if provided a `0`', function test( t ) {
+	var out;
+	var val;
+
+	out = [ 0, 0 ];
+	val = normalizef( 0.0, out, 1, 0 );
+	t.equal( val, out, 'returns output array' );
+	t.deepEqual( val, [0.0, 0], 'returns [0,0]' );
+	t.end();
+});
+
+tape( 'the function returns `[+inf,0]` if provided a `+infinity`', function test( t ) {
+	var out;
+	var val;
+
+	out = [ 0.0, 0.0 ];
+	val = normalizef( PINF, out, 1, 0 );
+	t.equal( val, out, 'returns output array' );
+	t.deepEqual( val, [PINF, 0], 'returns [+inf,0]' );
+	t.end();
+});
+
+tape( 'the function returns `[-inf,0]` if provided a `-infinity`', function test( t ) {
+	var out;
+	var val;
+
+	out = [ 0.0, 0.0 ];
+	val = normalizef( NINF, out, 1, 0 );
+	t.equal( val, out, 'returns output array' );
+	t.deepEqual( val, [NINF, 0], 'returns [-inf,0]' );
+	t.end();
+});
+
+tape( 'the function returns `[NaN,0]` if provided a `NaN`', function test( t ) {
+	var out;
+	var val;
+
+	out = [ 0.0, 0.0 ];
+	val = normalizef( NaN, out, 1, 0 );
+	t.equal( val, out, 'returns output array' );
+	t.equal( isnan( val[0] ), true, 'first element is NaN' );
+	t.equal( val[1], 0, 'second element is 0' );
+	t.end();
+});
+
+tape( 'the function supports providing an output array (array)', function test( t ) {
+	var out;
+	var val;
+
+	out = [ 3.14, 3.14 ];
+	val = normalizef( 0.0, out, 1, 0 );
+
+	t.strictEqual( val, out, 'returns output array' );
+	t.strictEqual( val[ 0 ], 0.0, 'first element is 0' );
+	t.strictEqual( val[ 1 ], 0, 'second element is 0' );
+
+	t.end();
+});
+
+tape( 'the function supports providing an output array (typed array)', function test( t ) {
+	var out;
+	var val;
+
+	out = new Float32Array( 2 );
+	out[ 0 ] = 3.14;
+	out[ 1 ] = 3.14;
+
+	val = normalizef( 0.0, out, 1, 0 );
+
+	t.strictEqual( val, out, 'returns output typed array' );
+	t.strictEqual( val[ 0 ], 0.0, 'first element is 0' );
+	t.strictEqual( val[ 1 ], 0, 'second element is 0' );
+
+	t.end();
+});
+
+tape( 'the function supports specifying a stride', function test( t ) {
```

Comment:
For this test and the next test, we should use a different test value. Providing `0` does not allow us to actually test whether providing stride or offset arguments actually affect behavior.